src/core/algorithms/ind/spider/attribute.h

```diff
@@ -0,0 +1,145 @@
+/** \file
+ * \brief Spider attribute
+ *
+ * Attribute class definition
+ */
+#pragma once
+
+#include <utility>
+#include <vector>
+
+#include <boost/dynamic_bitset.hpp>
+
+#include "config/error/type.h"
+#include "model/table/column_combination.h"
+#include "model/table/column_domain_iterator.h"
+#include "model/table/column_index.h"
+#include "util/bitset_utils.h"
+
+namespace algos::spider {
+
+using AttributeIndex = model::ColumnIndex;
+
+namespace details {
+/// base class for IND attributes
+class Attribute {
+public:
+    using Iterator = model::ColumnDomainIterator;
+
+protected:
+    AttributeIndex id_;         /* attribute unique identificator */
+    AttributeIndex attr_count_; /* attribute unique identificator */
+    Iterator it_;               /* domain iterator */
+
+public:
+    Attribute(AttributeIndex attr_id, AttributeIndex attr_count, model::ColumnDomain const& domain)
+        : id_(attr_id), attr_count_(attr_count), it_(domain) {}
+
+    /// get unqiue attribute id
+    AttributeIndex GetId() const {
+        return id_;
+    }
+
+    /// get attribute iterator
+    Iterator& GetIt() {
+        return it_;
+    }
+
+    /// get attribute iterator
+    Iterator const& GetIt() const {
+        return it_;
+    }
+
+    /// check whether the attribute has processed
+    virtual bool HasFinished() const {
+        return !GetIt().HasNext();
+    }
+
+    /// compare attributes first by their values and then by their ids
+    bool operator>(Attribute const& rhs) const {
+        int cmp = GetIt().GetValue().compare(rhs.GetIt().GetValue());
```

Comment:
Expression `GetIt().GetValue()` is used several times in different places, so maybe we should add method `GetValue()` to `Attribute` class?